### PR TITLE
Setup auth schemas

### PR DESCRIPTION
**Pull request by nkatzman**
This PR:
1) Sets up  bell, hapi-auth-cookie, hapi-auth-jwt auth schemes for use
2) Expose login route uses the OAUTH scheme. It will set a session cookie and return a jwt
3) Expose logout route that clears session cookie and returns nothing

The login route right now uses a whitelist in order to validate the username of users attempting to login

### CODE REVIEW

File: lib/login.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        notes: 'Authenticate user with github oauth provider',
+        tags: ['api', 'login'],
+        auth: {
+            strategy: 'oauth'
+        },
+        handler: (request, reply) => {
+            if (!request.auth.isAuthenticated) {
+                const message = `Authentication failed due to: ${request.auth.error.message}`;
+
+                return reply(boom.forbidden(message));
```
Comment (tkyi) [inline lib/login.js]: does `boom.unauthorized(message)` make more sense here

File: index.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+ * @param  {Function} next                          Function to call when done
+ */
+exports.register = (server, options, next) => {
+    server.register([bell, sugar, jwt], (err) => {
+        if (err) {
+            throw err;
+        }
+
+        const config = hoek.applyToDefaults(options, {
+            password: 'this_is_a_fake_password_that_should_be_overridden',
```
Comment (d2lam) [inline index.js]: Did you want it to be the other way around?

hoek.applyToDefaults({stuff.... }, options)

File: README.md [hunk 3]
```diff
@@ -9,6 +9,74 @@
...
+
+const Hapi = require('hapi');
+let server = new Hapi.Server();
+
+server.connection({ port: 8000 });
+
+server.register([{
+    register: require('screwdriver-plugin-login'),
+    options: {
+      password: 'this_is_a_password_that_needs_to_be_atleast_32_characters',
```
Comment (stjohnjohnson) [inline README.md]: 2 or 4 spaces

File: test/index.test.js [hunk 4]
```diff
@@ -1,8 +1,199 @@
 'use strict';
 const assert = require('chai').assert;
+// const sinon = require('sinon');
```
Comment (tkyi) [inline test/index.test.js]: remove commented code

**Comment by d2lam** [comment]
Sheep it! 🐑 
200w

**Comment by tkyi** [comment]
👍   :shipit:
giphy-2

**Pull request merged**